### PR TITLE
feat: add fields parent_content_key and is_assigned_course_run for assignments

### PR DESCRIPTION
**Pull request by adamstankiewicz**
**Description:**

* Adds 2 model fields to `LearnerContentAssignment`:
  * `parent_content_key`. This will represent a top-level course key when the assignment's `content_key` is a course run. Otherwise, `None`.
  * `is_assigned_course_run`. This will indicate whether the assignment is _course run_-based or _course_-based.
* Serializes the 2 new fields `parent_content_key` and `is_assigned_course_run` in API responses.
* Displays `is_assigned_course_run` as a read-only field in Django Admin.
* Implements helper method `_get_parent_content_key` to return either a course run's parent course's `content_key` or `None` if the provided `content_key` is already a course run.
* Updates assignment allocation to pass `parent_content_key` and `is_assigned_course_run` during assignment object creation.

**Jira:**
[ENT-8876](https://2u-internal.atlassian.net/browse/ENT-8876)

**Merge checklist:**
- [x] `./manage.py makemigrations` has been run
    - *Note*: This **must** be run if you modified any models.
      - It may or may not make a migration depending on exactly what you modified, but it should still be run.

**Post merge:**
- [ ] Ensure that your changes went out to the stage instance
- [ ] Deploy to prod instance

### CODE REVIEW

File: enterprise_access/apps/content_assignments/migrations/0023_historicallearnercontentassignment_is_assigned_course_run_and_more.py [hunk 1]
```diff
@@ -0,0 +1,39 @@
...
+        migrations.AddField(
+            model_name='learnercontentassignment',
+            name='parent_content_key',
+            field=models.CharField(blank=True, db_index=True, help_text='The globally unique content identifier of the parent content to assign to the learner.  Joinable with ContentMetadata.content_key in enterprise-catalog.', max_length=255, null=True),
+        ),
+        migrations.AlterField(
+            model_name='historicallearnercontentassignment',
+            name='allocated_at',
+            field=models.DateTimeField(blank=True, default=django.utils.timezone.now, help_text='The last time the assignment was allocated. Cannot be null.'),
+        ),
```
Comment (adamstankiewicz) [inline enterprise_access/apps/content_assignments/migrations/0023_historicallearnercontentassignment_is_assigned_course_run_and_more.py:38]: [inform] This migration is unrelated to changes I made in this PR; seems like it's a migration that was missing (i.e., auto-generated during `makemigrations`).

File: enterprise_access/apps/content_assignments/api.py [hunk 2]
```diff
@@ -560,14 +581,19 @@ def _create_new_assignments(
 
     # First, prepare assignment objects using data available in-memory only.
     content_title = _get_content_title(assignment_configuration, content_key)
+    parent_content_key = _get_parent_content_key(assignment_configuration, content_key)
     preferred_course_run_key = _get_preferred_course_run_key(assignment_configuration, content_key)
+    is_assigned_course_run = bool(parent_content_key)
```
Comment (adamstankiewicz) [inline enterprise_access/apps/content_assignments/api.py:586]: [inform] Uses a non-null `parent_content_key` to determine whether the assignment is run-based.

File: enterprise_access/apps/content_assignments/api.py [hunk 3]
```diff
@@ -525,6 +525,27 @@ def _get_content_title(assignment_configuration, content_key):
...
+def _get_parent_content_key(assignment_configuration, content_key):
+    """
+    Helper to retrieve (from cache) the parent content key of a content_key's content_metadata.
+    Note: content_key is either a course run key or a course key. Only course run keys have a
+    parent course key.
+
+    If content_key is for a course key, this will return the same key. Otherwise, the content_key
+    represents a course run, and this will return the run's parent course key.
+    """
+    content_metadata = _get_content_summary(assignment_configuration, content_key)
```
Comment (iloveagent57) [inline enterprise_access/apps/content_assignments/api.py:537]: nit: maybe name this `course_content_metadata` to make it clear that `_get_content_summary()` returns a course, and not a run, even if we pass a course run key into it.

**Review by brobro10000 — APPROVED** [review]
LGTM 👍🏽
